Solaris fcntl() returns 'Bad file number' when an exclusive lock is requested
on non-writable file.

This is a Solaris specific patch.

--- Django-3.2/django/test/testcases.py
+++ Django-3.2/django/test/testcases.py
@@ -1593,7 +1593,7 @@ class SerializeMixin:
             raise ValueError(
                 "{}.lockfile isn't set. Set it to a unique value "
                 "in the base class.".format(cls.__name__))
-        cls._lockfile = open(cls.lockfile)
+        cls._lockfile = open(cls.lockfile, 'r+')
         locks.lock(cls._lockfile, locks.LOCK_EX)
         super().setUpClass()
 
--- Django-3.2/tests/files/tests.py
+++ Django-3.2/tests/files/tests.py
@@ -171,6 +171,7 @@ class FileTests(unittest.TestCase):
             test_file.seek(0)
             self.assertEqual(test_file.read(), (content * 2).encode())
 
+    @unittest.skip("Case assumptions are not correct on Solaris")
     def test_exclusive_lock(self):
         file_path = Path(__file__).parent / 'test.png'
         with open(file_path) as f1, open(file_path) as f2:
